nautobot/extras/models/customfields.py

```diff
@@ -525,12 +526,52 @@ def to_form_field(
                 widget=DatePicker(),
             )
 
-        # Text and URL
-        elif self.type in (CustomFieldTypeChoices.TYPE_URL, CustomFieldTypeChoices.TYPE_TEXT):
+        # Text-like fields
+        elif self.type in (
+            CustomFieldTypeChoices.TYPE_URL,
+            CustomFieldTypeChoices.TYPE_TEXT,
+            CustomFieldTypeChoices.TYPE_MARKDOWN,
+            CustomFieldTypeChoices.TYPE_JSON,
+        ):
             if self.type == CustomFieldTypeChoices.TYPE_URL:
-                field = LaxURLField(required=required, initial=initial)
+                field = LaxURLField(
+                    required=required,
+                    initial=initial,
+                    min_length=self.validation_minimum,
+                    max_length=self.validation_maximum,
+                )
             elif self.type == CustomFieldTypeChoices.TYPE_TEXT:
-                field = forms.CharField(max_length=255, required=required, initial=initial)
+                field = forms.CharField(
+                    required=required,
+                    initial=initial,
+                    min_length=self.validation_minimum,
+                    max_length=self.validation_maximum,
+                )
+            elif self.type == CustomFieldTypeChoices.TYPE_MARKDOWN:
+                field = CommentField(
+                    widget=SmallTextarea,
+                    label=None,
+                    min_length=self.validation_minimum,
+                    max_length=self.validation_maximum,
+                )
+            elif self.type == CustomFieldTypeChoices.TYPE_JSON:
+                if simple_json_filter:
+                    field = JSONField(
+                        encoder=DjangoJSONEncoder,
+                        required=required,
+                        initial=None,
+                        widget=TextInput,
+                        min_length=self.validation_minimum,
+                        max_length=self.validation_maximum,
+                    )
+                else:
+                    field = JSONField(
+                        encoder=DjangoJSONEncoder,
+                        required=required,
+                        initial=initial,
+                        min_length=self.validation_minimum,
+                        max_length=self.validation_maximum,
+                    )
```

Comment:
Wonder if for the JSONField we shouldn't apply a min_length/max_length to the UI widget since the actual validation is enforced server-side and may differ depending on whitespace present?